commands/publish/index.js

```diff
@@ -84,6 +84,16 @@ class PublishCommand extends Command {
           `
         );
       }
+
+      GitUtilities.fetchGitRemote(this.execOpts);
+      if (GitUtilities.isBehindUpstream(this.gitRemote, this.execOpts)) {
```

Comment:
I'm wondering if there should be an option to either opt-in / opt-out of this check?

Reply:
On second thoughts, checking `gitEnabled && !skipNpm`  should be sufficient, as if you are publishing to npm, it is expected that there is a git remote to push tags to.

Reply:
At the same time, publishing to npm and pushing to git are (intended to be) separate concerns (even _if_ the implementation is currently kinda muddled). If the local checkout is behind upstream when making git commits and tags and pushing is enabled, it should always error, as that would inevitably error _anyway_ once the push to remote was attempted later in the process.

That is, _regardless_ of whether publishing to npm is enabled, if `this.gitEnabled === true` it is always an error if the upstream is ahead when publishing.